### DISCUSSION TITLE
Is location state reliable?

### DISCUSSION THREAD

**Discussion by Rc85** [remix-run/react-router#8919] [Q&A]
I'm currently passing a state object into history.push and utilizing it with the useLocation hook in the new page. I'm curious as to where the state is stored and is it reliable in terms of will it ever be lost from refresh, closing tab, closing browser, or any event? I'm doing this so that I don't have to make unncessary requests to my server for the same data that I already have from the page I'm navigating from.

**Comment by kiliman** [comment]
React Router uses the browser History API

Apparently Firefox stores history state on the users disk so it is recoverable after browser restarts, but I'm not sure I'd rely on that as it's most likely an implementation choice as the spec doesn't define anything about persistence.

I would assume that history state will at least live for the life of the browser session.

>The state object can be anything that can be serialized. Because Firefox saves state objects to the user's disk so they can be restored after the user restarts the browser, we impose a size limit of 16 MiB on the serialized representation of a state object. If you pass a state object whose serialized representation is larger than this to pushState(), the method will throw an exception. If you need more space than this, you're encouraged to use [sessionStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage) and/or [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage).

https://developer.mozilla.org/en-US/docs/Web/API/History/pushState

**Comment by thoth-seshat** [comment]
I'm pretty sure it's using something like this: https://developer.mozilla.org/en-US/docs/Web/API/History/state
